src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+public abstract class TwoPassVariantWalker extends VariantWalker {
+
+    @Override
+    public void traverse(){
+        final VariantFilter variantFilter = makeVariantFilter();
+        final CountingReadFilter readFilter = makeReadFilter();
+
+        // First pass through the variant
+        traverseVariants(variantFilter, readFilter, this::firstPassApply);
+        logger.info("Finished first pass through the variants");
+
+        afterFirstPass();
+
+        // Second pass
+        logger.info("Starting second pass through the variants");
+        traverseVariants(variantFilter, readFilter, this::secondPassApply);
+        logger.info(readFilter.getSummaryLine());
+    }
+
+
+    protected abstract void firstPassApply(final VariantContext variant,
```

Comment:
Add full javadoc for `firstPassApply()`, `secondPassApply()`, and `afterFirstPass()`, modeled after the javadoc for the analogous methods in `TwoPassReadWalker`